pkg/logs/launchers/integration/launcher.go

```diff
@@ -157,56 +253,117 @@ func (s *Launcher) makeFileSource(source *sources.LogSource, logFilePath string)
+	fileInfo := &FileInfo{
+		filename:     filepath,
+		lastModified: time.Now(),
+		size:         0,
+	}
+
+	return fileInfo, nil
 }
 
 // integrationLoglogFilePath returns a file path to use for an integration log file
 func (s *Launcher) integrationLogFilePath(id string) string {
-	fileName := strings.ReplaceAll(id, " ", "-")
-	fileName = strings.ReplaceAll(fileName, ":", "_") + ".log"
+	fileName := strings.ReplaceAll(id, ":", "_") + ".log"
 	logFilePath := filepath.Join(s.runPath, fileName)
 
 	return logFilePath
 }
 
-// ensureFileSize enforces the max file size for files integrations logs
-// files. Files over the set size will be deleted and remade.
-func (s *Launcher) ensureFileSize(logFilePath string) error {
-	maxFileSizeSetting := pkgConfig.Datadog().GetInt64("logs_config.integrations_logs_files_max_size")
-	maxFileSizeBytes := maxFileSizeSetting * 1024 * 1024
+// deleteAndRemakeFile deletes log files and creates a new empty file with the
+// same name
+func (s *Launcher) deleteAndRemakeFile(filepath string) error {
+	err := os.Remove(filepath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			ddLog.Warn("File does not exist, creating new one: ", err)
+		} else {
+			ddLog.Warn("Error deleting file: ", err)
+		}
+	} else {
+		ddLog.Info("Successfully deleted oversize log file, creating new one.")
+	}
 
-	fi, err := os.Stat(logFilePath)
+	file, err := os.Create(filepath)
 	if err != nil {
 		return err
 	}
+	defer file.Close()
+
+	return nil
+}
+
+// computerDiskUsageMax computes the max disk space the launcher can use based
+// off the integrations_logs_disk_ratio and integrations_logs_total_usage
+// settings
+func computeMaxDiskUsage(runPath string, logsTotalUsageSetting int64, usageRatio float64) (int64, error) {
+	usage, err := filesystem.NewDisk().GetUsage(runPath)
+	if err != nil {
+		return 0, err
+	}
+
+	diskReserved := float64(usage.Total) * (1 - usageRatio)
+	diskAvailable := int64(usage.Available) - int64(math.Ceil(diskReserved))
+
+	return min(logsTotalUsageSetting, diskAvailable), nil
+}
 
-	if fi.Size() > int64(maxFileSizeBytes) {
-		err := os.Remove(logFilePath)
+// scanInitialFiles scans the run path for initial files and then adds them to
+// be managed by the launcher
+func (s *Launcher) scanInitialFiles(dir string) error {
+	err := filepath.Walk(dir, func(_ string, info os.FileInfo, err error) error {
 		if err != nil {
-			if os.IsNotExist(err) {
-				ddLog.Warn("File does not exist, creating new one: ", err)
-			} else {
-				ddLog.Warn("Error deleting file: ", err)
-			}
-		} else {
-			ddLog.Info("Successfully deleted oversize log file, creating new one.")
+			return nil
 		}
 
-		file, err := os.Create(logFilePath)
-		if err != nil {
-			return err
+		if !info.IsDir() {
+			fileInfo := &FileInfo{
+				filename:     info.Name(),
+				size:         info.Size(),
+				lastModified: info.ModTime(),
+			}
+
+			integrationID := fileNameToID(fileInfo.filename)
+
+			s.integrationToFile[integrationID] = fileInfo
```

Comment:
What happens if: 
- the agent starts with integration ID: xyz
- it writes some logs
- the agent is stopped, and restarted without that integration anymore
- we will delete and recreate a file for ID: xyz

It seems like that file will remain on disk and tracked in the agent, empty, and stale forever. 
In this initial scan, should we just delete files, but not re-create them until the integration is scheduled? 
Perhapse we can add a test case to cover this scenario. 

Somthing to also keep in mind when you add support for integrations that are scheduled/unscheduled at runtime.

Reply:
Ahh that makes sense.

Reply:
I've refactored the launcher to have delete and make file functions now so they will be fully deleted instead of remade into a stale file. Also added a test for this scenario.